src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -598,14 +659,20 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
                                        layerThicknessEdge(k, iEdge)*gradDensityEdge(k, iEdge))/(layerThicknessEdge(k - 1, iEdge) + &
                                                                                                 layerThicknessEdge(k, iEdge))
                kappaGMEdge = 0.5_RKIND*(gmKappaScaling(k, cell1) + gmKappaScaling(k, cell2))
+               dzdxTopOfEdge = (layerThicknessEdge(k - 1, iEdge)*dzdxEdge(k - 1, iEdge) + &
+                                layerThicknessEdge(k, iEdge)*gradDensityEdge(k, iEdge))/(layerThicknessEdge(k - 1, iEdge) + &
+                                                                                            layerThicknessEdge(k, iEdge))
+
+               gradDensityConstZTopOfEdge = gradDensityTopOfEdge + dzdxTopOfEdge*drhoDzTopOfEdge(k, iEdge)
```

Comment:
I understand that this line is 
```
drho/dx = drho/dr + drho/dz*dz/dx
```
where x is the exact horizontal and r is the tilted direction. Because density depends strongly on depth, it's probably better to interpolate everything in temperature and salinity to the correct vertical location first (i.e. the same z location as the top of edge), and then compute
```
drho/dx = drho/dT*dT/dx + drho/dS*dS/dx
```
where x is the exact horizontal. I'll sketch out a picture and review the GM design document.

Reply:
@mark-petersen this makes sense.  Let me know what you see in the design doc.  I can make this change and put in a second test to see sensitivity.

Reply:
Qingshan Chen did a careful job on the derivation, years ago. He is showing
<img width="694" alt="image" src="https://user-images.githubusercontent.com/4036816/88215934-a1dbb280-cc19-11ea-99eb-f99dbde90455.png">
for this same line, which would mean that your line above should have a negative sign on the last term. I'm still looking into it. That design doc is here:
https://github.com/MPAS-Dev/MPAS-Documents/tree/master/ocean/gm
compiled version attached.